### PR TITLE
Add static internal catcher calculation methods for CatchDifficultyCalculator

### PR DESCRIPTION
**Pull request by frenzibyte**
Better and more simpler way IMO to calculate the width of the catching area without having to instantiate a whole `Catcher`.

### CODE REVIEW

File: osu.Game.Rulesets.Catch/UI/Catcher.cs [hunk 1]
```diff
@@ -139,6 +141,26 @@ private void load()
             updateCatcher();
         }
 
+        /// <summary>
+        /// Calculates the scale of the catcher based off the provided beatmap difficulty.
+        /// </summary>
+        internal static Vector2 CalculateScale(BeatmapDifficulty difficulty)
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/UI/Catcher.cs:147]: private?

File: osu.Game.Rulesets.Catch/UI/Catcher.cs [hunk 2]
```diff
@@ -139,6 +141,26 @@ private void load()
...
+        /// <summary>
+        /// Calculates the width of the area used for attempting catches in gameplay.
+        /// </summary>
+        internal static float CalculateCatchWidth(Vector2 scale)
+            => CatcherArea.CATCHER_SIZE * Math.Abs(scale.X) * allowed_catch_range;
+
+        /// <summary>
+        /// Calculates the width of the area used for attempting catches in gameplay.
+        /// </summary>
+        /// <param name="difficulty"></param>
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/UI/Catcher.cs:159]: missing xmldoc

File: osu.Game.Rulesets.Catch/UI/Catcher.cs [hunk 3]
```diff
@@ -139,6 +141,26 @@ private void load()
...
+        /// <summary>
+        /// Calculates the scale of the catcher based off the provided beatmap difficulty.
+        /// </summary>
+        internal static Vector2 CalculateScale(BeatmapDifficulty difficulty)
+            => new Vector2(1.0f - 0.7f * (difficulty.CircleSize - 5) / 5);
+
+        /// <summary>
+        /// Calculates the width of the area used for attempting catches in gameplay.
+        /// </summary>
+        internal static float CalculateCatchWidth(Vector2 scale)
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/UI/Catcher.cs:153]: missing param xmldoc, no idea what scale is.

**Review by peppy — DISMISSED** [review]

File: osu.Game.Rulesets.Catch/UI/Catcher.cs (hunk 1 above)
Comment (iiSaLMaN) [reply osu.Game.Rulesets.Catch/UI/Catcher.cs:147]: it's a method for calculating same as the others so wasn't sure about making it private, will do.

Comment (peppy) [reply osu.Game.Rulesets.Catch/UI/Catcher.cs:147]: if a method is not used externally (and doesn't carry any foreseeable purpose) then it should be private.

File: osu.Game.Rulesets.Catch/UI/Catcher.cs [hunk 4]
```diff
@@ -139,6 +141,26 @@ private void load()
...
+        /// Calculates the scale of the catcher based off the provided beatmap difficulty.
+        /// </summary>
+        private static Vector2 calculateScale(BeatmapDifficulty difficulty)
+            => new Vector2(1.0f - 0.7f * (difficulty.CircleSize - 5) / 5);
+
+        /// <summary>
+        /// Calculates the width of the area used for attempting catches in gameplay.
+        /// </summary>
+        /// <param name="scale">The scale of the catcher.</param>
+        internal static float CalculateCatchWidth(Vector2 scale)
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Catch/UI/Catcher.cs:154]: Overload is probably not required but ¯\\_(ツ)_/¯

**Review by smoogipoo — APPROVED** [review]

File: osu.Game.Rulesets.Catch/UI/Catcher.cs (hunk 4 above)
Comment (iiSaLMaN) [reply osu.Game.Rulesets.Catch/UI/Catcher.cs:154]: I think you meant the other one: `CalculateCatchWidth(BeatmapDifficulty)` (which is there for simplifying `Catcher.CalculateCatchWidth(Catcher.CalculateScale(difficulty))`). 

this one is required to be as-is for working with calculating the catch-width without a `BeatmapDifficulty`:
https://github.com/ppy/osu/blob/883788dd5aec1b228328812fb4ea14043c299118/osu.Game.Rulesets.Catch/UI/Catcher.cs#L111

Comment (smoogipoo) [reply osu.Game.Rulesets.Catch/UI/Catcher.cs:154]: Nope, I mean't this one. As in I would've been fine just recomputing the scale a second time in all cases.

**Pull request merged**